### PR TITLE
remove executable stack from hcc shared libraries

### PR DESCRIPTION
**Pull request by david-salinas**
Change-Id: Ib3d10b0cfb97cf737f571c49156c304d92c2401e
for issue #1056

### CODE REVIEW

File: scripts/cmake/MCWAMP.cmake [hunk 1]
```diff
@@ -27,7 +28,7 @@ macro(amp_target name )
   target_include_directories(${name} PRIVATE ${MCWAMP_INC_DIR})
   target_include_directories(${name} SYSTEM INTERFACE $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include>)
   target_compile_options(${name} PUBLIC -hc -fPIC)
-
+
```
Comment (aaronenyeshi) [inline scripts/cmake/MCWAMP.cmake]: there was a random trailing space added here

**Review by aaronenyeshi — COMMENTED** [review]
What is the context for this change?

File: scripts/cmake/MCWAMP.cmake (hunk 1 above)
Comment (david-salinas) [reply scripts/cmake/MCWAMP.cmake]: doh.  ok will remove.  And there is another one at line 71.

**Comment by david-salinas** [comment]
> What is the context for this change?

@aaronenyeshi the context is that by default when we build libmcwamp_hsa.so, we leave in an executable stack in the SO.  Issue #1056 is the original complaint.  This is actually, just default behaviour for clang and the linker.  But because our SO gets linked in to user/customer executables, this executable section can introduce a security vulnerability.  So we can just add the linker option "noexecstack" to remove the "X" (execute) permission from the "GNU_STACK" entry in the elf.

**Comment by scchan** [comment]
@david-salinas I don't think this PR addresses #1056 since this only addresses the shared libraries produced by HCC's cmake file.  #1056's concern is about objects produced by HCC so I think the fix should be somewhere's HCC's linker script.

**Pull request closed without merging**